kafka-rest/src/main/java/io/confluent/kafkarest/controllers/SchemaManager.java

```diff
@@ -26,12 +27,42 @@
   /**
    * Returns the {@link RegisteredSchema schema} registered with the given {@code schemaId}.
    */
-  RegisteredSchema getSchemaById(EmbeddedFormat format, int schemaId);
+  RegisteredSchema getSchemaById(String subject, int schemaId);
```

Comment:
schema:id is 1:1 - why do you need a subject? this looks wrong

Reply:
Although schema ID uniquely identifies the actual schema, we also return the schema version in `RegisteredSchema`. To acquire schema version, we need schema ID and subject. This also guarantees the subject has a version with this ID. Otherwise serialization would fail later.